### ISSUE TITLE
details to achieve the paper results

### ISSUE THREAD

**Issue by lei-liu1** [nguyentthong/CLNTM#1]
Hello!
Thanks for your great job!

I have some questions about the details of the results.
I want to replicate the paper results and run the code like this:
`python run_scholar.py data/20ng/processed/ -k 50 --topk 15 --test-prefix test --device 0 --o ./outputs/scholar/20ng_50_15 --epochs 500 --model trueContrastiveScholar --dist 0`
and get result 0.3298 +- 0.0024 (5 runs)，a little lower than your result 0.334 +- 0.004.

I find that your code is based on the Scholar model, so I wonder if you have incorporated the metadata, such as labels, covariates, and/or word2vec embeddings. If so, how did you use them? 
Thanks all.

**Comment by nguyentthong** [comment]
Dear @lei-liu1,

Thank you for your interest in our work. Regarding the incorporation of metadata, you can execute the following commands (with the additional --labels tag)

`python run_scholar.py data/20ng/processed/ -k 50 --topk 15 --test-prefix test --device 0 --o ./outputs/scholar/20ng_50_15 --epochs 500 --model trueContrastiveScholar --dist 0 --labels sentiment`

Regards,
Thong

**Comment by lei-liu1** [comment]
Thanks a lot!

**Comment by lei-liu1** [comment]
Hello, @nguyentthong

I can not replicate the paper results on IMDB and Wiki datasets. I read the code carefully and found a bug in `scholar.py` `torchTrueContrastiveScholar` class, you should have added the contrastive loss in `_loss` method as in `torchContrastiveScholar` class. I fixed this, but can not achieve the paper results.

I run the code like this:
`python run_scholar.py data/imdb/processed/ --dist 0 --topk 15 -k 50 -l 0.001 --w2v GoogleNews-vectors-negative300.bin.gz --test-prefix test --device 0 --o ./outputs/scholar/imdb_50_15 --epochs 500 --model trueContrastiveScholar --labels sentiment`
`python run_scholar.py data/wiki/processed/ -k 50 --topk 15 -l 0.002 --test-prefix test --device 0 --o ./outputs/scholar/wiki_50_15 --epochs 500 --model trueContrastiveScholar --batch-size 500 --alpha 0.01 --dist 0`

I have adjusted the parameters, such as learning rate, epoch numbers, with and without word2vec, but there is still a big gap between our results (e.g. 0.145 v.s. 0.197 on IMDB dataset). I want to know if there is something I have not considered yet.

I use python3.6.5 and pytorch1.9.1.

Looking forward to your reply. 
Thanks!
